src/main/java/net/sf/jabref/BibDatabaseContext.java

```diff
@@ -65,7 +65,7 @@ public BibDatabaseContext(BibDatabase database, MetaData metaData, File file) {
         this(database, metaData, file, new Defaults());
     }
 
-    public BibDatabaseContext(Defaults defaults, DatabaseLocation location, String keywordSeparator) {
+    public BibDatabaseContext(Defaults defaults, DatabaseLocation location, Character keywordSeparator) {
```

Comment:
There were some issues related to strings as keyword separators. For example, the default is `, ` (with space) but some places tried to split strings only at the comma (i.e. recognize `apple,orange`). The space was only used to reformat the list as `apple, orange`. So I decided to force that only one character can be used as a separator and the final space is always adapted upon writing.